atom/browser/net/proxying_url_loader_factory.cc

```diff
@@ -32,6 +34,16 @@ void ProxyingURLLoaderFactory::CreateLoaderAndStart(
     const network::ResourceRequest& request,
     network::mojom::URLLoaderClientPtr client,
     const net::MutableNetworkTrafficAnnotationTag& traffic_annotation) {
+  // Check if user has intercepted this scheme.
+  auto it = protocol_->intercept_handlers().find(request.url.scheme());
+  if (it != protocol_->intercept_handlers().end()) {
+    AtomURLLoaderFactory(it->second.first, it->second.second)
```

Comment:
Any reason to create this factory on the stack ?

Reply:
To call `CreateLoaderAndStart` we have to create the factory first, and to create the factory creating on stack is the simplest and cheapest way, calling `make_unique` would allocate memory on heap which is unnecessarily expensive.

Reply:
In that case, we can maintain one `AtomURLLoaderFactory` per network context and make new connections using `AtomURLLoaderFactory::Clone` which will essentially reference count the class and delete it when all the connection drops. `AppCacheSubresourceURLFactory` is a good example.

Reply:
Didn't know that trick, will do some learning on it.

On this code specially, I realize we don't even need to create `AtomURLLoaderFactory` at all, I have changed the code to call the helper method directly to avoid any extra overhead.